### PR TITLE
ST-303, ST-1096: Add compare products page

### PR DESCRIPTION
**Pull request by ilyawzrd**
Description:

--

QA-test: ST-306

Demo-test: ST-307

Download artifact URL: https://vc3prerelease.blob.core.windows.net/packages/client-app-0.0.4-pr-89-51dcf524.zip

### CODE REVIEW

File: client-app/pages/compare-products/compare-products.vue [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+              </div>
+            </div>
+            <AddToCart :product="product" />
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+<script setup lang="ts">
```
Comment (Flanker72) [inline client-app/pages/compare-products/compare-products.vue:113]: Divide template and script by 1 empty line

File: client-app/shared/catalog/components/empty-comparison.vue [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  <div class="max-w-screen-2xl px-5 md:px-12 mx-auto">
+    <VcBreadcrumbs :items="breadcrumbs"></VcBreadcrumbs>
+    <h1 class="text-black-800 text-3xl uppercase mt-2 font-bold">
+      {{ $t("pages.compare.empty_list.title") }}
+    </h1>
+  </div>
+  <TwoColumn class="max-w-screen-lg">
+    <template #left>
+      <div class="flex flex-col h-full py-12 justify-center items-center lg:items-start">
+        <VcImage src="/static/images/errors/emptyCart.webp" alt="emptyCartMobile" class="lg:hidden mb-8" />
```
Comment (Flanker72) [inline client-app/shared/catalog/components/empty-comparison.vue:11]: Wrong alt attribute, please use localized string

File: client-app/shared/catalog/components/empty-comparison.vue [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        <h2 class="text-black-800 text-2xl font-bold mb-8 text-center md:text-left">
+          {{ $t("pages.compare.empty_list.message") }}
+        </h2>
+        <VcButton :to="`/${SeoUrl.Catalog}`" size="lg" class="p-4 uppercase font-bold">{{
+          $t("pages.compare.empty_list.button_text")
+        }}</VcButton>
+      </div>
+    </template>
+    <template #right>
+      <VcImage src="/static/images/errors/emptyCart.webp" alt="emptyCart" class="w-full max-w-md" />
```
Comment (Flanker72) [inline client-app/shared/catalog/components/empty-comparison.vue:21]: Wrong alt attribute, please use localized string

File: client-app/shared/catalog/components/empty-comparison.vue [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+<script setup lang="ts">
+import { VcImage, VcBreadcrumbs, VcButton, IBreadcrumbs } from "@/components";
+import { TwoColumn } from "@/shared/layout";
+import SeoUrl from "@core/seo-routes.enum";
+import { ref } from "vue";
+
+const breadcrumbs = ref<IBreadcrumbs[]>([
+  { title: "Home", url: "/" },
+  { title: "Compare products", url: "/compare-products" },
```
Comment (Flanker72) [inline client-app/shared/catalog/components/empty-comparison.vue:34]: Breadcrumbs are not localized

Comment (ilyawzrd) [reply client-app/shared/catalog/components/empty-comparison.vue]: fixed

File: client-app/pages/compare-products/compare-products.vue (hunk 1 above)
Comment (ilyawzrd) [reply client-app/pages/compare-products/compare-products.vue:115]: fixed

File: client-app/pages/compare-products/compare-products.vue [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+import _ from "lodash";
+import { onMounted, ref, watch } from "vue";
+import { breakpointsTailwind, useBreakpoints } from "@vueuse/core";
+import { useCompareProducts } from "@/shared/compare";
+import { Product as ProductType } from "@/core/api/graphql/types";
+const { fetchProducts, products } = useProducts();
+const { clearCompareList, productsLimit, removeFromCompareList, productsIds } = useCompareProducts();
+
+const breadcrumbs = ref<IBreadcrumbs[]>([
+  { title: "Home", url: "/" },
```
Comment (Flanker72) [inline client-app/pages/compare-products/compare-products.vue]: Need to localize breadcrumbs

File: client-app/pages/compare-products/compare-products.vue [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+  { title: "Compare products", url: "/compare-products" },
+]);
+
+const breakpoints = useBreakpoints(breakpointsTailwind);
+const isMobile = breakpoints.smaller("lg");
+
+const showOnlyDifferences = ref(false);
+
+const originalProperties = ref<{ [key: string]: { value: string }[] }>({});
```
Comment (Flanker72) [inline client-app/pages/compare-products/compare-products.vue:140]: Why not to use type or interface here and on the next line?

File: client-app/shared/catalog/components/empty-comparison.vue (hunk 3 above)
Comment (ilyawzrd) [reply client-app/shared/catalog/components/empty-comparison.vue]: done

File: client-app/shared/catalog/components/empty-comparison.vue (hunk 2 above)
Comment (ilyawzrd) [reply client-app/shared/catalog/components/empty-comparison.vue]: done

File: client-app/pages/compare-products/compare-products.vue [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+      }
+    });
+  } else {
+    computedProperties.value = { ...originalProperties.value };
+  }
+};
+
+function getProductProperties() {
+  if (_.isEmpty(products.value)) return;
+  const grouped: { [key: string]: { value: string }[] } = {};
```
Comment (Flanker72) [inline client-app/pages/compare-products/compare-products.vue:170]: This is the same type as on the above comment, better to use type or interface instead

Comment (ilyawzrd) [reply client-app/pages/compare-products/compare-products.vue]: done

File: client-app/pages/compare-products/compare-products.vue [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+        <div
+          class="items-start md:items-center space-x-5 px-5 lg:px-0 border-b border-gray-100 lg:border-0"
+          v-for="(values, key, index) in computedProperties"
+          :key="index"
+          :class="!isMobile ? 'even:bg-gray-50 flex' : productsIds.length >= 3 ? 'inline-flex' : 'flex'"
+        >
+          <div class="w-1/6 pl-8 font-extrabold text-sm" v-if="!isMobile">{{ key }}</div>
+          <div class="w-32 flex-shrink-0 lg:flex-shrink md:w-48 py-5" v-for="(value, index) in values" :key="index">
+            <span v-if="isMobile" class="block font-extrabold text-sm">{{ key }}</span>
+            {{ value.value }}
```
Comment (Flanker72) [inline client-app/pages/compare-products/compare-products.vue:91]: Better to use `property.value` and renave v-for argument above.

Comment (ilyawzrd) [reply client-app/pages/compare-products/compare-products.vue]: done

**Review by Flanker72 — APPROVED** [review]

**Pull request merged**